lib/puppet/transaction/report.rb

```diff
@@ -386,6 +394,101 @@ def self.default_format
 
   private
 
+  # Using the last applied values from our persistence store, and the current
+  # report data, iterate across all events finding any corrective_change that
+  # might exist and marking the event, resource and report appropriately.
+  def process_resource_inferences
+    persistence = Puppet::Transaction::Persistence.new
+    persistence.load
+
+    new_resources = {}
+
+    resource_statuses.each do |name, status|
+      # Initialize data structure for resource
+      new_resources[name] ||= {}
+      params = new_resources[name]["parameters"] ||= {}
+
+      # Populate new hash with events mapped to params
+      param_to_event = {}
+      status.events.each do |ev|
+        param_to_event[ev.property] = ev
+      end
+
+      status.real_resource.parameters.each do |pname,param|
+        # Skip if if the data is sensitive, or if its just a property
+        next if !param.is_a?(Puppet::Property) || param.sensitive
```

Comment:
Is there a reason we're not using `#select` to pick the non-sensitive properties and then iterate over those instead of iterating over all and selectively skipping the loops? `next` is kind of a funny thing and I would rather we avoid using it.